internal/decryption/header.go

```diff
@@ -34,6 +34,10 @@ type HeaderRecord struct {
 }
 
 func (h *HeaderRecord) DecryptKey() ([]byte, error) {
+	err := h.KeyProvider.ValidateCredentials()
+	if err != nil {
```

Comment:
```suggestion
	if err := h.KeyProvider.ValidateCredentials(); err != nil {
```
since we don't use the err outside of this block...